app/models/page.rb

```diff
@@ -38,4 +42,11 @@ def url_must_have_domain
       errors.add(:url, 'must have a domain')
     end
   end
+
+  def sync_titles(version)
+    if version.title.present?
+      most_recent_capture_time = versions.limit(1).order(capture_time: :desc).pluck(:capture_time).first
+      self.title = version.title if most_recent_capture_time.nil? || most_recent_capture_time <= version.capture_time
```

Comment:
You need to also call `save!` here to save this change to the DB. You could also call `update` instead of `title=` (`update` immediately saves):

```rb
update(title: version.title) if most_recent_capture_time.nil? || most_recent_capture_time <= version.capture_time
```

See below comment in the tests about making sure the changes are saved, too.